### ISSUE TITLE
solve ui_metrics challenge during login

### ISSUE THREAD

**Issue by JulienMaille** [iSarabjitDhiman/TweeterPy#89]
Thanks a lot for your repository and the port of Go implementation of `X-Twitter-Active-User` header generator.
I noticed that during login flow, there's another challenge related to `/i/js_inst?c_name=ui_metrics`
The same genius wrote about this: https://github.com/obfio/twitter-ui_mentrics
Have you had time to give it a look? That may help reduce `DenyLoginSubtask` issues, what do you think?

**Comment by iSarabjitDhiman** [comment]
Hi @JulienMaille 
Yes I checked this one a while ago. But this implementation is not stable, it may break anytime I believe. I was looking for a permanent solution, thats why I kind of left it. Though I have implemented it on my side for personal use, but I used headless mode of selenium. I will implement some functionality to solve ui_metrics soon, just looking for a better solution. If I don't find one, I will surely borrow this one.

**Comment by JulienMaille** [comment]
Hi, what have you found unstable about it? I ported to code to python and could verify it generates the expected result. Which part of the implementation do you dislike?

**Comment by iSarabjitDhiman** [comment]
No, I mean yeah the code works seamlessly. But I was looking for a robust solution rather than using regex to replace parts of JavaScript code with python native syntax. I would prefer writing the whole thing in python. Of course we need regex for some part, but I would avoid it as much I can.
But again, I am more than happy to implement the same approach if I don't find any other solution.

Hopefully I will fix it by the end of this month (in a week).

**Comment by iSarabjitDhiman** [comment]
Thanks for sharing.
Yes it's as same as the one written in Go. How long has this one in working condition? Did you ever had to modify or fix it? Just wondering when did u write it, how long has it been?

**Comment by JulienMaille** [comment]
It's fresh from yesteraday

**Comment by iSarabjitDhiman** [comment]
Ok thanks. I will find a solution and implement it soon.

**Comment by d60** [comment]
hello,
I have published a script that solves ui_metrics using js2py. 
https://github.com/d60/twitter_ui_metrics
